libs/zard/src/lib/components/tabs/tabs.component.ts

```diff
@@ -178,21 +200,42 @@ export class ZardTabGroupComponent {
   public readonly zScrollAmount = input(100);
   public readonly zAlignTabs = input<zAlign>('start');
 
-  constructor() {
-    afterNextRender(() => {
-      if (this.tabs().length > 0) {
-        this.setActiveTab(0);
-      }
+  protected readonly showArrow = computed(() => this.zShowArrow() && this.scrollPresent());
+
+  ngAfterViewInit(): void {
+    // default tab selection
+    if (this.tabs().length) {
+      this.setActiveTab(0);
+    }
+
+    runInInjectionContext(this.injector, () => {
+      const observeInputs$ = merge(toObservable(this.zShowArrow), toObservable(this.tabs));
+      afterNextRender(() => {
+        const resizeObserver = new ResizeObserver(() => {
+          this.setScrollState();
+        });
+        resizeObserver.observe(this.tabsContainer().nativeElement);
 
-      combineLatest([interval(100).pipe(startWith(0))]).subscribe(() => {
-        this.hasScrollSignal.set(this.hasScroll());
+        merge(observeInputs$, fromEvent(window, 'resize'))
+          .pipe(debounceTime(10), takeUntilDestroyed(this.destroyRef))
+          .subscribe(() => this.setScrollState());
+
+        this.destroyRef.onDestroy(() => {
+          resizeObserver.disconnect();
+        });
       });
     });
   }
 
+  private setScrollState(): void {
+    if (this.hasScroll() !== this.scrollPresent()) {
+      this.scrollPresent.set(this.hasScroll());
+    }
+  }
+
   private hasScroll(): boolean {
-    if (this.tabsContainer && this.tabsContainer().nativeElement && this.zShowArrow()) {
-      const navElement: HTMLElement = this.tabsContainer().nativeElement;
+    const navElement: HTMLElement = this.tabsContainer().nativeElement;
+    if (this.zShowArrow()) {
```

Comment:
I'm not sure if `navElement` will always have a value.

```suggestion
    if (this.zShowArrow() && navElement) {
```

Reply:
Perfect, thanks for explaining that.